gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -30,26 +31,25 @@
 
     private CodeableConceptMapper codeableConceptMapper;
 
-    public ProcedureRequest mapToProcedureRequest(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04PlanStatement planStatement) {
+    public ProcedureRequest mapToProcedureRequest(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04PlanStatement planStatement,
+        Patient patient) {
 
         /**
          * TODO: Known future implementations to this mapper
-         * - subject: references a global patient resource for the transaction (NIAD-2024)
          * - context: references an encounter resource if it has been generated from the ehrComposition (NIAD-2025)
          * - requester: fallback to a default 'Unknown User' Practitioner if none are present in requester (NIAD-2026)
          * - concatenate source practice org id to identifier URL (NIAD-2021)
          */
 
         var id = planStatement.getId().getRoot();
-        var identifier = getIdentifier(id);
         var note = getNote(planStatement.getText());
         var reasonCode = codeableConceptMapper.mapToCodeableConcept(planStatement.getCode());
         var authoredOn = getAuthoredOn(planStatement.getAvailabilityTime(), ehrExtract, planStatement.getId());
         var occurrence = getOccurrenceDate(planStatement.getEffectiveTime());
         var agentReference = ParticipantReferenceUtil.getParticipantReference(planStatement.getParticipant(),
             EhrResourceExtractorUtil.extractEhrCompositionForPlanStatement(ehrExtract, planStatement.getId()));
 
-        return createProcedureRequest(id, identifier, note, reasonCode, authoredOn, occurrence, agentReference);
+        return createProcedureRequest(id, note, reasonCode, authoredOn, occurrence, agentReference, new Reference(patient));
```

Comment:
new Reference(patient) will create a Reference object with Patient object inside, 
don't we rather want to create a reference by the IdType? (patient.getIdElement() passed as param)

Reply:
I think this is the simplest way of setting a reference - HAPI library handles everything for us. What is the advantage of passing patient.getIdElement() ?